### PR TITLE
Add checks, flags, and help to publish.sh

### PR DESCRIPTION
**Pull request by WillNilges**
Basically, a full re-write of the current publishing script. I think it's more trouble than it's worth to infer things, and since this script is given to the user above the game directory, it's very counter-intuitive to have to move it as soon as you get it.

What changed?
- Required flag arguments for image, banner, and game directory
- Checks MIME type for png
- Checks image dimensions
- Checks that the game directory is actually a directory
- Copies artifacts into temporary directory
- Cleans up after itself. The only artifact left over should be your zip file.

### CODE REVIEW

File: publish.sh [hunk 1]
```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+# Correct dimensions for banner
+CORRECT_B_W=800
+CORRECT_B_H=450
+
+# Correct dimensions for icon
+CORRECT_I_W=128
+CORRECT_I_H=128
```
Comment (WillNilges) [inline publish.sh]: Open to deliberation about this. Could do a multiple, or just make sure that it's square.

**Comment by WillNilges** [comment]
Would love some feedback about the help message and the docstrings inside the script. Are they clear enough? Is it obvious what is happening at a glance?

File: publish.sh (hunk 1 above)
Comment (MTFT-Games) [reply publish.sh]: I think just ensuring that it is square is quite good enough. If you want to change it to that... or i could change it to that... hmmmm

**Comment by MTFT-Games** [comment]
I think the help and comments look pretty dang good. Merging...

**Pull request merged**